### ISSUE TITLE
ws-daemon: cannot set CPU limit

### ISSUE THREAD

**Issue by jenting** [gitpod-io/gitpod#12329]
### Bug description

From GCP log in us63, we see the `cannot set CPU limit` warning.

```console
failed to read cpu max from /mnt/node-cgroups/kubepods.slice/kubepods-burstable.slice/kubepods-burstable-podcb18ade6_d0d0_4628_aee6_7ac23bf77760.slice/cri-containerd-003849cea86b7ff9ce11c71668530cb61181da817427c2be89ce3fa8bc08c078.scope:
    github.com/gitpod-io/gitpod/ws-daemon/pkg/cpulimit.CgroupV2CFSController.SetLimit
        github.com/gitpod-io/gitpod/ws-daemon/pkg/cpulimit/cfs_v2.go:35
  - unable to read cpu.max:
    github.com/gitpod-io/gitpod/ws-daemon/pkg/cpulimit.CgroupV2CFSController.readCpuMax
        github.com/gitpod-io/gitpod/ws-daemon/pkg/cpulimit/cfs_v2.go:65
  - open /mnt/node-cgroups/kubepods.slice/kubepods-burstable.slice/kubepods-burstable-podcb18ade6_d0d0_4628_aee6_7ac23bf77760.slice/cri-containerd-003849cea86b7ff9ce11c71668530cb61181da817427c2be89ce3fa8bc08c078.scope/cpu.max: no such file or directory
```

### Steps to reproduce

Dunno

**Comment by utam0k** [comment]
@jenting This is a natural error. The message is that the pod has already gone, but the tick has been run due to a timing issue. Was this an error? What is probably expected is a warning level.

**Comment by jenting** [comment]
Thank you @utam0k for the quick response.

> Was this an error?
Only this one is indicated as a Warning.

If you think it's fine, feel free to close this issue.

**Comment by utam0k** [comment]
@jenting Thanks for your heads-up

**Comment by jenting** [comment]
> @jenting This is a natural error. The message is that the pod has already gone, but the tick has been run due to a timing issue. Was this an error? What is probably expected is a warning level.

https://github.com/gitpod-io/gitpod/blob/a817faca8dd5db4725e6aa9a3458c11b13ed2bf3/components/ws-daemon/pkg/cpulimit/dispatch.go#L155-L165

We did the workspace exists check at line 155. I'm wondering if this happens the workspace pod gone immediately after line 155? Or, we did not handle the `d.workspaces` correctly even the workspace pod was gone.
